src/cogent3/core/sequence.py

```diff
@@ -2946,51 +2946,51 @@ class ArrayProteinWithStopSequence(ArraySequence):
 
 
 @singledispatch
-def _coerce_seq(data, preserve_case, checker):
+def _coerce_seq(data, seqid, preserve_case, checker):
     from cogent3.core.alignment import Aligned
 
     if isinstance(data, Aligned):
-        return _coerce_seq(str(data), preserve_case, checker)
+        return _coerce_seq(str(data), seqid, preserve_case, checker)
```

Comment:
`_coerce_seq` should probably be called `_coerce_to_seqview()`, since that's what it's doing. So, we may as well simplify some of these where we know we can a `SeqView` instance directly. Hence, change
```python
return _coerce_seq(str(data), seqid, preserve_case, checker)
```
to
```python
return _coerce_seq(data.data, seqid, preserve_case, checker)
```
(which then goes to the Sequence case)